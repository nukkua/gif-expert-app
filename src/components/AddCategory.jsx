import { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ onAddCategory }) => {
	// estamos evitando el two way databinding??
	const [inputValue, setInputValue] = useState('');

	const onInputChange = (event) => {
		setInputValue(event.target.value);
	};

	const onSubmit = (event) => {
		const newValueSanitized = inputValue.trim();
		event.preventDefault();
		if (newValueSanitized.length <= 1) {
			return;
		}
		onAddCategory(newValueSanitized);
		setInputValue('');
	};

	return (
		<>
			<form onSubmit={onSubmit} aria-label="form">
				<input value={inputValue} onChange={onInputChange} type="text" placeholder="Buscar gifs" />
			</form>
		</>
	);
};

AddCategory.protoTypes = {
	onAddCategory: PropTypes.func.isRequired
};
