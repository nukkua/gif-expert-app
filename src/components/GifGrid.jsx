import { GifItem } from './GifItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({ category }) => {
	const { images, isLoading } = useFetchGifs(category);
	return (
		<>
			{isLoading && <h2>Cargando...</h2>}
			<h3>{category}</h3>
			<div className="card-grid">
				{images.map((image) => (
					<GifItem key={image.id} {...image} />
				))}
			</div>
		</>
	);
};

//NOTA
// El operador spread para cuando tengo un chingo de props como esta en el <GifItem>

// useEffect => tiene 2 parametros un callback, y la lista de dependencias
// la lista de dependencias vacia, me dice que solamente se va renderizar una vez
// lo que este dentro del callback

// NOTA
// Yo no necesito toda la data de cada objeto en el respuesta right?
// Graphql me permitiria seleccionar que campos quiero obtener
// Pero como no estamos usando Graphql, usaremos la funcion getGifs
// Que aparte de hacer el fetch, sanitiza los objetos

// OTRA NOTA ANASHE
// La cosa es que el <React.StrictMode> hace que la wea se renderize 2 veces,
// por eso vemos 2 consolelog del array de gifs
