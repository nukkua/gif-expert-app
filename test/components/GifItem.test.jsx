import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('first test', () => {
	const gif = {
		title: 'One punch man',
		url: 'https://media2.giphy.com/media/X8cs1EFGQgVL5bhVDH/giphy.gif?cid=4720ee8dr73naon14e3aqfihvo8hvm4k48hgwhw3sdtnn4k0&ep=v1_gifs_search&rid=giphy.gif&ct=g'
	};
	test('Matching Snapshot testing', () => {
		const { container } = render(<GifItem {...gif} />);
		expect(container).toMatchSnapshot();
	});
	test('Debe de matchear el url con el que tengo en mi constante gif', () => {
		render(<GifItem {...gif} />);
		// expect(screen.getByRole('img').src).toBe(gif.url);
		// expect(screen.getByRole('img').alt).toBe(gif.title);
		const { src, alt } = screen.getByRole('img');
		expect(src).toBe(gif.url);
		expect(alt).toBe(gif.title);
	});
	test('Debe de matchear el titulo su innerHTML con el nuestro', () => {
		render(<GifItem {...gif} />);
		expect(screen.getByText(gif.title).innerHTML).toBeTruthy();
	});
});
