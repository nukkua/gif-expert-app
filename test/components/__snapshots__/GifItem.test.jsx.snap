// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`first test Matching Snapshot testing 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="One punch man"
      src="https://media2.giphy.com/media/X8cs1EFGQgVL5bhVDH/giphy.gif?cid=4720ee8dr73naon14e3aqfihvo8hvm4k48hgwhw3sdtnn4k0&ep=v1_gifs_search&rid=giphy.gif&ct=g"
    />
    <p>
      One punch man
    </p>
  </div>
</div>
`;
